File: src/test/java/com/indiepost/repository/ContributorRepositoryTests.java
package com.indiepost.repository;

import com.indiepost.NewIndiepostApplicationKt;
import com.indiepost.enums.Types;
import com.indiepost.model.Contributor;
import org.apache.commons.lang3.RandomStringUtils;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Sort;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.test.context.web.WebAppConfiguration;

import javax.inject.Inject;
import javax.transaction.Transactional;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import static org.assertj.core.api.Java6Assertions.assertThat;

@RunWith(SpringRunner.class)
@SpringBootTest(classes = NewIndiepostApplicationKt.class)
@WebAppConfiguration
@Transactional
public class ContributorRepositoryTests {

    @Inject
    private ContributorRepository contributorRepository;

    private List<Long> insertedId = new ArrayList<>();


    @Before
    public void beforeTests() {
        Contributor contributor = new Contributor();
        contributor.setFullName(RandomStringUtils.randomAlphanumeric(10));
        contributor.setEmail("before@example.com");
        contributor.setSubEmail("test_before_sub@example.com");
        contributor.setDescription("Hello World!");
        contributor.setDisplayType(Types.ContributorDisplayType.TEXT);
        contributor.setTitle("Test Before title");
        contributor.setContributorType(Types.ContributorType.FreelanceEditor);

        LocalDateTime now = LocalDateTime.now();
        contributor.setLastUpdated(now);
        contributor.setCreated(now);

        Long id = contributorRepository.save(contributor);
        insertedId.add(id);
    }

    @Test
    public void save_shouldSaveEntityAndReturnIdProperly() {
        Contributor contributor = new Contributor();
        contributor.setFullName(RandomStringUtils.randomAlphanumeric(10));
        contributor.setEmail("test@example.com");
        contributor.setSubEmail("test_sub@example.com");
        contributor.setDescription("Hello World");
        contributor.setDisplayType(Types.ContributorDisplayType.TEXT);
        contributor.setTitle("Test Title");
        contributor.setContributorType(Types.ContributorType.FeatureEditor);

        LocalDateTime now = LocalDateTime.now();
        contributor.setLastUpdated(now);
        contributor.setCreated(now);

        Long id = contributorRepository.save(contributor);
        if (id != null) {
            insertedId.add(id);
        }
        assertThat(id).isNotNull();
    }

    @Test
    public void delete_shouldDeleteContributorProperly() {
        List<Contributor> contributors = contributorRepository.findAll(
                PageRequest.of(0, 10, Sort.Direction.DESC, "id")
        ).getContent();
        if (contributors.size() > 0) {
            for (Contributor contributor : contributors) {
                contributorRepository.delete(contributor);
            }
        }
        contributors = contributorRepository.findAll(
                PageRequest.of(0, 10, Sort.Direction.DESC, "id")).getContent();
        assertThat(contributors.size()).isEqualTo(0);
    }

    @After
    public void afterTests() {
        for (Long id : insertedId) {
            contributorRepository.deleteById(id);
        }
    }
}

File: src/test/kotlin/com/indiepost/repository/ContributorRepositoryTests.kt
package com.indiepost.repository

import com.indiepost.NewIndiepostApplication
import com.indiepost.enums.Types
import com.indiepost.model.Contributor
import org.apache.commons.lang3.RandomStringUtils
import org.assertj.core.api.Java6Assertions.assertThat
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Sort
import org.springframework.test.context.junit.jupiter.SpringExtension
import org.springframework.test.context.web.WebAppConfiguration
import java.time.LocalDateTime
import java.util.*
import javax.inject.Inject
import javax.transaction.Transactional

@ExtendWith(SpringExtension::class)
@SpringBootTest(classes = [NewIndiepostApplication::class])
@WebAppConfiguration
@Transactional
class ContributorRepositoryTests {

    @Inject
    private lateinit var contributorRepository: ContributorRepository

    private val insertedId = ArrayList<Long>()

    @BeforeEach
    fun beforeTests() {
        val contributor = Contributor()
        contributor.fullName = RandomStringUtils.randomAlphanumeric(10)
        contributor.email = "before@example.com"
        contributor.subEmail = "test_before_sub@example.com"
        contributor.description = "Hello World!"
        contributor.displayType = Types.ContributorDisplayType.TEXT
        contributor.title = "Test Before title"
        contributor.contributorType = Types.ContributorType.FreelanceEditor

        val now = LocalDateTime.now()
        contributor.lastUpdated = now
        contributor.created = now

        val id = contributorRepository.save(contributor)
        insertedId.add(id)
    }

    @Test
    fun save_shouldSaveEntityAndReturnIdProperly() {
        val contributor = Contributor()
        contributor.fullName = RandomStringUtils.randomAlphanumeric(10)
        contributor.email = "test@example.com"
        contributor.subEmail = "test_sub@example.com"
        contributor.description = "Hello World"
        contributor.displayType = Types.ContributorDisplayType.TEXT
        contributor.title = "Test Title"
        contributor.contributorType = Types.ContributorType.FeatureEditor

        val now = LocalDateTime.now()
        contributor.lastUpdated = now
        contributor.created = now

        val id = contributorRepository.save(contributor)
        if (id != null) {
            insertedId.add(id)
        }
        assertThat(id).isNotNull()
    }

    @Test
    fun delete_shouldDeleteContributorProperly() {
        var contributors = contributorRepository.findAll(
                PageRequest.of(0, 10, Sort.Direction.DESC, "id")
        ).content
        if (contributors.size > 0) {
            for (contributor in contributors) {
                contributorRepository.delete(contributor)
            }
        }
        contributors = contributorRepository.findAll(
                PageRequest.of(0, 10, Sort.Direction.DESC, "id")).content
        assertThat(contributors.size).isEqualTo(0)
    }

    @AfterEach
    fun afterTests() {
        for (id in insertedId) {
            contributorRepository.deleteById(id)
        }
    }
}
